src/Microsoft.EntityFrameworkCore.Relational.Specification.Tests/WarningsTestBase.cs

```
@@ -80,7 +80,62 @@ public virtual void Single_SingleOrDefault_without_orderby_doesnt_issue_warning(
                 Assert.NotNull(query2);
             }
         }
-      
+
+        [Fact]
+        public virtual void Last_with_order_by_does_not_issue_client_eval_warning()
```

We should always client eval Last because in case collection is empty it should throw an exception. LastOrDefault we could translate safely